frontend/src/features/admin-form/settings/SettingsMultiLangPage.tsx

```diff
@@ -1,15 +1,40 @@
 import { useSearchParams } from 'react-router-dom'
+import _ from 'lodash'
+
+import { TRANSLATION_INPUT, UNICODE_LOCALE } from '~constants/routes'
 
 import { MultiLanguageSection } from './components/MultiLanguageSection/MultiLanguageSection'
 import { TranslationListSection } from './components/MultiLanguageSection/TranslationListSection'
+import { TranslationSection } from './components/MultiLanguageSection/TranslationSection'
 
 export const SettingsMultiLangPage = (): JSX.Element => {
   const [searchParams] = useSearchParams()
-  const unicodeLocale = searchParams.get('unicodeLocale')
+  const unicodeLocale = searchParams.get(UNICODE_LOCALE)
+  const translationInput = searchParams.get(TRANSLATION_INPUT)
+  const isTranslationInput = !_.isNull(translationInput)
+  const isEndPageTranslationInput = translationInput === 'endPage'
+  const isStartPageTransltionInput = translationInput === 'startPage'
 
   return (
```

Comment:
Here's my attempt at separating out the the nested ternary operators out from the render function -- I generally avoid deeply nested render functions as they can be hard to read.

```suggestion
  // Request user to select a language
  if (!unicodeLocale) {
    return <MultiLanguageSection />
  }

  // Request user to select fields to translate
  if (!isTranslationInput) {
    return <TranslationListSection language={unicodeLocale} />
  }

  return (
    <TranslationSection
      language={unicodeLocale}
      formFieldNumToBeTranslated={
        isEndPageTranslationInput || isStartPageTransltionInput
          ? -1
          : _.toNumber(translationInput)
      }
      isEndPageTranslations={isEndPageTranslationInput}
      isStartPageTranslations={isStartPageTransltionInput}
    />
  )
```

Reply:
Thanks for this suggestion. It's a really neat way of handling multiple logics.